### ISSUE TITLE
CharsReaderObservableSuite has a flaky test

### ISSUE THREAD

**Issue by Avasil** [monix/monix#1030]
It sometimes fails on CI with 
```
[info] - fromCharsReader closes the file handle on cancel *** FAILED ***
[info]   received Some(Success(())) != expected None (CharsReaderObservableSuite.scala:181)
```

e.g. in https://travis-ci.org/monix/monix/jobs/585428074

**Comment by pk044** [comment]
it wouldn't change anything if we changed the sleep duration to something longer, right?

i wonder if it would make sense to see what data makes this test fail? and then see if its reproducible locally?

**Comment by Avasil** [comment]
> it wouldn't change anything if we changed the sleep duration to something longer, right?

I don't think so, it uses `TestScheduler` and `s.tick()` doesn't allow any time to pass

> i wonder if it would make sense to see what data makes this test fail? and then see if its reproducible locally?

I suppose we could create a PR which prints generated input and rerun CI until it fails

**Comment by alexandru** [comment]
I just fixed it in #1122 

The problem is that `randomReaderWithOnFinish`, uses `randomString()`, which can return an empty string, yielding an empty `Observable`.
